web/src/click_handlers.ts

```diff
@@ -939,11 +939,28 @@ export function initialize(): void {
         }
 
         if (compose_state.composing() && $(e.target).parents("#compose").length === 0) {
-            const is_inside_link = $(e.target).closest("a").length > 0;
-            if (is_inside_link || $(e.target).closest(".copy_codeblock").length > 0) {
+            const click_is_inside_link = $(e.target).closest("a").length > 0;
+            const anchor_parent_ele = document.getSelection()?.anchorNode?.parentElement;
+            const focus_parent_ele = document.getSelection()?.focusNode?.parentElement;
+
+            // If the selection extends outside a link, no need to prevent the click behavior.
+            const selection_is_inside_link =
+                anchor_parent_ele?.closest("a") === focus_parent_ele?.closest("a") &&
+                anchor_parent_ele?.closest("a") !== null;
+
+            const clicked_link = $(e.target).closest("a").get(0);
+            const selected_link = anchor_parent_ele?.closest("a");
+            // We want to prevent the default click behavior only if the selection and click
+            // lies inside the same link.
+            const should_prevent_click_behavior =
+                click_is_inside_link && selection_is_inside_link && selected_link === clicked_link;
+            if (
+                should_prevent_click_behavior ||
+                $(e.target).closest(".copy_codeblock").length > 0
+            ) {
                 // We want to avoid blurring a selected link by triggering a
                 // focus event on the compose textarea.
-                if (is_inside_link && document.getSelection()?.type === "Range") {
+                if (click_is_inside_link && document.getSelection()?.type === "Range") {
                     // To avoid the click behavior if a link is selected.
                     e.preventDefault();
                     return;
```

Comment:
Would just using `mouse_drag.is_drag` fix this? That only consider selections that were plausibly caused by a drag.